src/Pages/UserProfile/UserProfile.js

```diff
@@ -0,0 +1,201 @@
+import "./UserProfile.css";
+import { BatNetworkContainer } from "../../Components/Wrapper/BatNetworkContainer";
+import { Sidebar } from "../../Components/Sidebar/Sidebar";
+import { RightSidebar } from "../../Components/RightSidebar/RightSidebar";
+import { useNavigate } from "react-router-dom";
+import { ButtonSecondary, ButtonPrimary } from "../../Components/Buttons";
+import { UserListModal, UserProfileEditModal } from "../../features/user";
+import { useParams } from "react-router-dom";
+import {
+  getPosts,
+  resetPostsOfUser,
+  getPostsOfUser,
+  PostCard,
+} from "../../features/post";
+import { getAllUsers } from "../../features/user";
+import { logoutHandler } from "../../features/auth";
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
```

Comment:
Create barrels so you can import these from one single common file to keep the code concise!